packages/core/src/response-synthesizers/factory.ts

```diff
@@ -77,6 +77,16 @@ class Refine extends BaseSynthesizer {
     }
   }
 
+  async getResponse(
+    query: MessageContent,
+    nodes: NodeWithScore[],
+    stream: true,
+  ): Promise<AsyncIterable<EngineResponse>>;
+  async getResponse(
+    query: MessageContent,
+    nodes: NodeWithScore[],
+    stream: false,
+  ): Promise<EngineResponse>;
```

Comment:
Added these signatures so that I didn't have to do more type jumbling after grabbing the original response and transforming it. If this is not accurate please let me know, but based on my knowledge if `stream = true` it will always be an AsyncIterable.